pkg/services/store/kind/registry.go

```diff
@@ -0,0 +1,134 @@
+package kind
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/store/kind/dummy"
+	"github.com/grafana/grafana/pkg/services/store/kind/playlist"
+)
+
+type KindRegistry interface {
+	Register(info models.ObjectKindInfo, builder models.ObjectSummaryBuilder) error
+	GetSummaryBuilder(kind string) models.ObjectSummaryBuilder
+	GetInfo(kind string) (models.ObjectKindInfo, error)
+	GetKinds() []models.ObjectKindInfo
+}
+
+func NewKindRegistry() KindRegistry {
+	kinds := make(map[string]*kindValues)
+	kinds[models.StandardKindPlaylist] = &kindValues{
+		info:    playlist.GetObjectKindInfo(),
+		builder: playlist.GetObjectSummaryBuilder(),
+	}
+
+	// FIXME -- these are registered because existing tests use them
+	for _, k := range []string{"kind1", "kind2", "kind3"} {
+		kinds[k] = &kindValues{
+			info:    dummy.GetObjectKindInfo(k),
+			builder: dummy.GetObjectSummaryBuilder(k),
+		}
+	}
+
+	// create a registry
+	reg := &registry{
+		mutex: sync.RWMutex{},
+		kinds: kinds,
+	}
+	reg.updateInfoArray()
+	return reg
+}
+
+// Zero dependency service -- this includes the default services
+func ProvideService() KindRegistry {
+	return NewKindRegistry()
+}
```

Comment:
i've already gone through all of the work of making a zero dependency generated registry for coremodels. i'm quite concerned about introducing a manual one.

Reply:
um wire?   and I thought we agree that kinds != coremodel?

Also, the internal kinds may have dependencies (svg - and currently datasource lookup for dashboard parsing)

Reply:
IIUC, this `KindRegistry` will be `coremodels` + `extra`, where extra can be either raw types, or structured entities that do not yet have a thema schema. Is that right?